### PR TITLE
Dashboards: Fixes performance issue expanding a row 

### PR DESCRIPTION
**Pull request by torkelo**
When expanding a row with many panels on a dashboard with many collapsed rows it resulted in A LOT of calls to detectdetectSaveModelChanges (which has a very expensive step inside sceneToSaveModel which does a sorting of all properties in the resulting save model) 

image

This freezes the browser for several seconds. We must have missed to debounce this function (we do it in panel editor) but not here. 

To simplify the unit tests I skip the debounce in unit tests. 

Related: https://github.com/grafana/support-escalations/issues/13059

### CODE REVIEW

File: public/app/features/dashboard-scene/saving/DashboardSceneChangeTracker.ts [hunk 1]
```diff
@@ -120,13 +121,8 @@ export class DashboardSceneChangeTracker {
...
   }
 
-  private onStateChanged(event: SceneObjectStateChangedEvent) {
-    if (DashboardSceneChangeTracker.isUpdatingPersistedState(event)) {
-      this.detectSaveModelChanges();
-    }
-  }
-
   private detectSaveModelChanges() {
+    console.log('detectSaveModelChanges');
```
Comment (dprokop) [inline public/app/features/dashboard-scene/saving/DashboardSceneChangeTracker.ts:125]:
```suggestion
```

**Review by dprokop — APPROVED** [review]

**Review by axelavargas — APPROVED** [review]
The code looks good to me and works as expected 🏅

**Comment by ivanortegaalba** [comment]
@torkelo maybe we can use [the worker](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard-scene/saving/DetectChangesWorker.ts), to detect the changes. This will make this async, but it would leave the main thread free and avoid the blocking. We're only using the worker now to highlight the save changes button, but this could be another use case. Wdyt?

**Comment by torkelo** [comment]
@ivanortegaalba We already are, we can only send serialized objects to the worker and it's this serialization that happens when we build the message to the worker that is freezing the browser. 

Or do you have something else in mind?

**Comment by ivanortegaalba** [comment]
> @ivanortegaalba We already are, we can only send serialized objects to the worker and it's this serialization that happens when we build the message to the worker that is freezing the browser.
> 
> Or do you have something else in mind?

Ups, yes, you're right. The serialization work is the hard part. I checked if there is any way to solve this. The main problem is the circular references because postMessage detects that they are not clonable. I haven't tried with SharedArrayBuffer, with no success. There are alternatives, but I didn't spend the time to avoid the serialization.

**Pull request merged**